### PR TITLE
Feature/visa

### LINKED ISSUE #677: Add Visa to Ego DB
## Detailed Description
Create tables for Visas, a java repo class to fetch Visas.  
Make migrate scripts to create visa. 

Refer to Bashar's NOtes:
Sql DB Changes
```
CREATE TABLE GA4GHVISA (
  id                      UUID PRIMARY KEY,
  type                    ENUM,
  source                  varchar(255) NOT NULL,
  value                   varchar(255) NOT NULL,
  by                      varchar(255) NOT NULL,
);


```

### LINKED ISSUE #678: Create Admin CRUD Endpoints for Visa
## Detailed Description
1. endpoints to read visa by visa ID, create, update, delete visas.
2. 1 endpoint to get all visas.
3. All of the endpoints should:
- Only accessible to admins
- have basic error handling
- have pagination

### PR DESCRIPTION
**Pull request by Azher2Ali**
Created a new develop branch in order to merge the changes related to Visa endpoints and the Visa Schema changes.

Possible Enhancement- Convert VisaType fields from String to ENUM
#677 #678

### CODE REVIEW

File: src/main/java/bio/overture/ego/controller/VisaController.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    return visaService.create(visaRequest);
+  }
+
+  @AdminScoped
+  @RequestMapping(method = PUT, value = "/{id}")
+  @ApiResponses(value = {@ApiResponse(code = 200, message = "Update Visa", response = Visa.class)})
+  public @ResponseBody Visa updateVisa(
+      @ApiIgnore @RequestHeader(value = "Authorization", required = true)
+          final String authorization,
+      @RequestBody(required = true) VisaUpdateRequest visaRequest) {
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/controller/VisaController.java]: a @PathVariable UUID **id** is missing in updateVisa as a parameter.
The PUT mapping requires an /{id}  !!

File: src/main/java/bio/overture/ego/model/dto/VisaUpdateRequest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class VisaUpdateRequest {
+
+  @NotNull private UUID id;
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/model/dto/VisaUpdateRequest.java]: should we be able to update Visa id? if not, we can remove id field.

File: src/main/java/bio/overture/ego/model/dto/VisaRequest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class VisaRequest {
+
+  private UUID id;
```
Comment (leoraba) [inline src/main/java/bio/overture/ego/model/dto/VisaRequest.java]: shouldn't be the **ID** autogenerated by the database on visa creation?

File: src/main/java/bio/overture/ego/model/dto/VisaUpdateRequest.java (hunk 2 above)
Comment (Azher2Ali) [reply src/main/java/bio/overture/ego/model/dto/VisaUpdateRequest.java]: Removed UUID from update request structure

**Review by leoraba — APPROVED** [review]
👍

**Pull request merged**